### PR TITLE
#9Task. Ihar Tsykala

### PR DESCRIPTION
**Pull request by IharTsykala**
https://travis-ci.org/github/IharTsykala/js-assignments/builds/711827226

### CODE REVIEW

File: task/09-functions-n-closures-tasks.js [hunk 1]
```diff
@@ -25,11 +24,12 @@
  *   getComposition(Math.sin, Math.asin)(x) => Math.sin(Math.acos(x))
  *
  */
-function getComposition(f,g) {
-    throw new Error('Not implemented');
+function getComposition(f, g) {
+  return function (x) {
```
Comment (KostyaKov) [inline task/09-functions-n-closures-tasks.js]: A simpler implementation would be through an arrow function in one line

File: task/09-functions-n-closures-tasks.js [hunk 2]
```diff
@@ -104,10 +109,17 @@ function memoize(func) {
  * retryer() => 2
  */
 function retry(func, attempts) {
-    throw new Error('Not implemented');
+  return function () {
+    while (attempts !== 0) {
+      try {
+        return func()
+      } catch (e) {
+        // console.log('I is in TV))')
```
Comment (KostyaKov) [inline task/09-functions-n-closures-tasks.js]: According to the example, an error must be thrown in the "catch" block

File: task/09-functions-n-closures-tasks.js [hunk 3]
```diff
@@ -132,10 +144,20 @@ function retry(func, attempts) {
  *
  */
 function logger(func, logFunc) {
-    throw new Error('Not implemented');
+  return function () {
+    let funcName = `${func.name}(${JSON.stringify(Array.from(arguments)).slice(
+      1,
```
Comment (KostyaKov) [inline task/09-functions-n-closures-tasks.js]: In this case, the "slice" method parameters will be more readable on the same line.

**Review by KostyaKov — COMMENTED** [review]
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by IharTsykala** [comment]
Please, accept my pr

**Comment by KostyaKov** [comment]
Please add and push fixes to the task according to the comments.
I'll mark the PR as "Draft", please click "ready for review" when it will be finished. Thank you!

**Comment by IharTsykala** [comment]
I excluded changes in exist code and comment. Please, accept my pr

https://travis-ci.org/github/IharTsykala/js-assignments/builds/713275119

**Comment by IharTsykala** [comment]
I excluded changes in exist code and comments. Please, accept my pr

https://travis-ci.org/github/IharTsykala/js-assignments/builds/713989002

**Comment by IlyaMokin** [comment]
Looks good, thanks! The PR is accepted

**Pull request closed without merging**